`xargo doc` doesn't seem to work

Xargo is working fabulously for my embedded project, with one exception.

```
[cbiffle@anansi emb1]$ xargo build --release
    Finished release [optimized + debuginfo] target(s) in 0.0 secs
[cbiffle@anansi emb1]$ xargo doc --release
 Documenting bitflags v0.7.0
error[E0463]: can't find crate for `core`

error: aborting due to previous error

error: Could not document `bitflags`.

To learn more, run the command again with --verbose.
error: `cargo` process didn't exit successfully
```

`xargo doc` doesn't seem to be hunting for crates in the sysroot.  Am I doing something wrong?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,12 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+### Fixed
+
+- `xargo doc`, which wasn't working because we didn't pass --sysroot to rustdoc. Note that rustdoc
+  gained support for '--sysroot' as of nightly-2016-06-28, so that version or newer is required to
+  use `xargo doc`.
+
 ## [v0.1.9]
 
 ### Fixed

--- a/src/main.rs
+++ b/src/main.rs
@@ -47,7 +47,7 @@ fn run(config_opt: &mut Option<Config>) -> CargoResult<()> {
                          is not set")
         }));
 
-    let (mut cargo, target, verbose) = try!(parse_args());
+    let (mut cargo, target, verbose, is_cargo_doc) = try!(parse_args());
 
     let rustflags = &try!(rustflags(config));
     let profiles = &try!(parse_cargo_toml(config));
@@ -73,6 +73,10 @@ fn run(config_opt: &mut Option<Config>) -> CargoResult<()> {
             } else {
                 cargo.env("RUSTFLAGS",
                           format!("{} --sysroot={}", rustflags.join(" "), sysroot));
+            }
+
+            if is_cargo_doc {
+                cargo.env("RUSTDOCFLAGS", format!("--sysroot={}", sysroot));
             }
         }
 
@@ -142,8 +146,9 @@ impl Target {
     }
 }
 
-fn parse_args() -> CargoResult<(Command, Option<Target>, bool)> {
+fn parse_args() -> CargoResult<(Command, Option<Target>, bool, bool)> {
     let mut cargo = Command::new("cargo");
+    let mut is_cargo_doc = false;
     let mut target = None;
     let mut verbose = false;
 
@@ -167,12 +172,16 @@ fn parse_args() -> CargoResult<(Command, Option<Target>, bool)> {
             if arg == "-v" || arg == "--verbose" {
                 verbose = true;
             }
+
+            if arg == "doc" {
+                is_cargo_doc = true;
+            }
         }
 
         cargo.arg(arg_os);
     }
 
-    Ok((cargo, target, verbose))
+    Ok((cargo, target, verbose, is_cargo_doc))
 }
 
 /// Returns the RUSTFLAGS the user has set either via the env variable or via build.rustflags

--- a/tests/smoke.rs
+++ b/tests/smoke.rs
@@ -99,6 +99,26 @@ fn simple() {
     cleanup(TARGET);
 }
 
+#[test]
+fn doc() {
+    const TARGET: &'static str = "__doc";
+
+    let td = try!(TempDir::new("xargo"));
+    let td = &td.path();
+    try!(try!(File::create(td.join(format!("{}.json", TARGET)))).write_all(CUSTOM_JSON.as_bytes()));
+
+    run(xargo().args(&["init", "--vcs", "none", "--name", TARGET]).current_dir(td));
+    try!(try!(OpenOptions::new().truncate(true).write(true).open(td.join("src/lib.rs")))
+         .write_all(LIB_RS));
+    run(xargo().args(&["doc", "--target", TARGET]).current_dir(td));
+
+    for krate in CRATES {
+        assert!(exists_rlib(krate, TARGET));
+    }
+
+    cleanup(TARGET);
+}
+
 // Calling `xargo build` twice shouldn't trigger a sysroot rebuild
 // The only case the sysroot would have to be rebuild is when the source is updated but that
 // shouldn't happen when running this test suite.

